compute_endpoint/globus_compute_endpoint/engines/base.py

```diff
@@ -251,6 +252,7 @@ def submit(
         -------
         future
         """
+        assert self._engine_ready, "Engine not started and cannot execute tasks"
```

Comment:
We should raise a standard exception (e.g., `RuntimeError`) instead of use an `assert` statement, which is meant for debugging purposes.

Reply:
Fixed.